src/WrappedMToken.sol

```diff
@@ -689,6 +688,48 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
         emit StoppedEarning(account_);
     }
 
+    /**
+     * @dev   Transfer `amount_` M to `recipient_`, tracking this contract's M balance rounding errors.
+     * @param recipient_ The account to transfer M to.
+     * @param amount_    The amount of M to transfer.
+     */
+    function _transferM(address recipient_, uint240 amount_) internal {
+        uint240 startingBalance_ = _mBalanceOf(address(this));
+
+        // NOTE: The behavior of `IMTokenLike.transfer` is known, so its return can be ignored.
+        IMTokenLike(mToken).transfer(recipient_, amount_);
+
+        // NOTE: When this WrappedMToken contract is earning, any amount of M sent from it is converted to a principal
+        //       amount at the MToken contract, which when represented as a present amount, may be a rounding error
+        //       amount more than `amount_`. In order to capture the real decrease in M, the difference between the
+        //       ending and starting M balance is returned.
+        uint240 decrease_ = startingBalance_ - _mBalanceOf(address(this));
+
+        // If the M lost is more than the wM burned, then the difference is added to `roundingError`.
+        roundingError += int144(int256(uint256(decrease_)) - int256(uint256(amount_)));
```

Comment:
Can you remind me the rationale for using a signed integer here? 

If the WrappedM contract is not earning, then `decrease == amount`.

Therefore, rounding errors are only possible when WrappedM is earning:
| Recipient Earning | Amount -> Principal | Amount Transferred | Rounding Error |
| ----------------- | -------------------- | -------------------- | --------------- |
| true | rounds up in favor of protocol | >= expected | decrease >= amount |
| false | rounds up in favor of protocol | >= expected | decrease >= amount |

Reply:
If both the sender and recipient are earners, the amount of principal taken from the sender is rounded up and that same rounded up principal is added to the recipient, so the recipient may get more. So, if an M earner wraps they M, the recipient (the Wrapaper contract) may gain more M.